SPECS/plug-late-sr.spec

```
@@ -0,0 +1,34 @@
+Summary: Retry plugging PBDs to specific SRs on boot
+Name: plug-late-sr
+Version: 1.0
+Release: 1%{?dist}
+License: GPLv3
+Source0: plug-late-sr
+Source1: plug-late-sr.service
+BuildArch: noarch
+
+Requires: python2-configparser
+
+%description
+Retry plugging PBDs to specific SRs on boot.
+
+%install
+install -D -m 755 %{SOURCE0} %{buildroot}%{_bindir}/plug-late-sr
+install -D -m 644 %{SOURCE1} %{buildroot}%{_unitdir}/plug-late-sr.service
+
+%post
+systemctl enable plug-late-sr.service
```

```suggestion
%systemd_post plug-late-sr.service
```

This evaluates to:
```
if [ $1 -eq 1 ] ; then 
        # Initial installation 
        systemctl preset plug-late-sr.service >/dev/null 2>&1 || : 
fi
```

... which will enable the service if the preset file doesn't explictly say it should be disabled, and also fixes three issues:
* unit should only be enabled once, at first install, not each time we update the RPM
* filtered output
* always returns true, as RPM scriptlets almost always should

---

I would like to enable the service by default. This package is specific and if you install it, you want to protect your SRs. Otherwise I don't see a good reason to install the RPM with this behavior:

``` 
# Disable the service on a fresh XCP-ng install...
systemctl preset plug-late-sr.service
```

---

As I wrote, systemctl preset should enable the service by default. It doesn't?

---

Ok, so uoting https://www.freedesktop.org/software/systemd/man/systemd.preset.html:

> If no preset files exist, preset operations will enable all units that are installed by default. If this is not desired and all units shall rather be disabled, it is necessary to ship a preset file with a single, catchall "disable *" line. (See example 1, below.)

There *is* a catchall "disable *" in `/usr/lib/systemd/system-preset/99-default-disable.preset`, so we need indeed to enable the service.

So my suggestion becomes:
```
if [ $1 -eq 1 ] ; then 
        # Initial installation: enable the service
        systemctl enable plug-late-sr.service >/dev/null 2>&1 || : 
fi
```

If this RPM weren't for a workaround script, we'd look into updating the default presets in `xcp-ng-release-presets` so that the service's preset value is `enabled`.